_posts/2016-04-12-ecto-model-schema-extension.md

```diff
@@ -0,0 +1,124 @@
+---
+layout: post
+cover: 'assets/images/general-cover-3.jpg'
+title: Ecto Model Schema Extension
+tags: docs
+subclass: 'post tag-docs'
+categories: 'elixir'
+author: 'Pikender'
+navigation: true
+logo: 'assets/images/nectar-cart.png'
+---
+
+>
+The post belongs to _NectarCommerce and Extension Framework Awareness_ Series
+>
+1. _[NectarCommerce Vision](http://vinsol.com/blog/2016/04/08/nectarcommerce-vision/)_
+1. _[Extension Framework Game Plan](http://vinsol.com/blog/2016/04/12/extension-framework-game-plan/)_
+1. Intro to Metaprogramming
+1. **Ecto Model Schema Extension**
+1. Ecto Model Support Functions Extension
+1. Phoenix Router Extension
+1. Phoenix View Extension
+1. Running Multiple Elixir Apps Together
+1. Extension Approach Explained
+1. Developer Experience and Workflow developing Favorite Product Extension
+1. Developer Experience and Workflow testing Favorite Product Extension
+
+
+## What will be NectarCommerce
+
+>
+Off-the-shelf Opensource E-commerce application for building an online store.
+>
+Provides an Extension Framework to support features not included in core as extensions.
+>
+Strives for unobtrusive parallel development of NectarCommerce and Extensions
+
+NectarCommerce is committed to providing a ready-to-use e-commerce solution but the definition of 100% is different under different business domains. It aims to solve common use-cases as part of the project and relying on extension framework to tap the rest.
+
+# Ecto Model Schema Extension
+
+### Why
+
+We want to allow Extensions to modify the schema of existing Nectar Models without changing the Nectar Models.
+
+Extensions should be able to add new fields and associations to existing models as needed for the cause.
```

Comment:
Extensions should be able to add new fields and associations to existing models **as and when needed**.